### PR TITLE
mass matrix between two field PtrList

### PR DESCRIPTION
**Pull request by vressegu**

### CODE REVIEW

File: src/ITHACA_CORE/ITHACAutilities/ITHACAcoeffsMass.C [hunk 1]
```diff
@@ -83,12 +83,15 @@ Eigen::MatrixXd getMassMatrix(
...
+    Eigen::MatrixXd M = Eigen::MatrixXd::Zero(Msize,Msize);
 
     if (consider_volumes)
     {
         Eigen::VectorXd V = getMassMatrixFV(modes[0]);
-        M = F.transpose().topRows(Msize) * V.asDiagonal() * F.leftCols(Msize);
+        for (int i=0; i<V.size(); i++)
+        {
+          M += V(i) * F.transpose().topRows(Msize).col(i) * F.leftCols(Msize).row(i);
+        }
```
Comment (giovastabile) [inline src/ITHACA_CORE/ITHACAutilities/ITHACAcoeffsMass.C:102]: Why this change? I think the vectorized version should be considerably faster.

File: src/ITHACA_CORE/ITHACAutilities/ITHACAcoeffsMass.C [hunk 2]
```diff
@@ -116,6 +119,78 @@ template Eigen::MatrixXd getMassMatrix(
...
+             "The Number of requested modes is larger then the available quantity.");
+
+    Eigen::MatrixXd F = Foam2Eigen::PtrList2Eigen(modes);
+    Eigen::MatrixXd F2 = Foam2Eigen::PtrList2Eigen(modes2);
+    Eigen::MatrixXd M = Eigen::MatrixXd::Zero(Msize,Msize2);
+
+    if (consider_volumes)
+    {
+        Eigen::VectorXd V = getMassMatrixFV(modes[0]);
+        for (int i=0; i<V.size(); i++)
```
Comment (giovastabile) [inline src/ITHACA_CORE/ITHACAutilities/ITHACAcoeffsMass.C]: I think also this one can be written as the previous one

File: src/ITHACA_CORE/ITHACAutilities/ITHACAcoeffsMass.C (hunk 1 above)
Comment (vressegu) [reply src/ITHACA_CORE/ITHACAutilities/ITHACAcoeffsMass.C:102]: I think that I had some memory issue for large mesh (mesh size ~ 10^6).
I will double check.
And anyway, I should probably do something that does not slow down the algorithm in the "small mesh case".

**Comment by vressegu** [comment]
Florian corrected this merge request.
For data of moderate size (e.g. volScalarField on a mesh with less than 1e6 cells), your (faster) vectorized version is used.
For larger data, the slower version with the loop is used, solving RAM issues.

**Comment by vressegu** [comment]
I close this merge request and I will create a new one, from a dedicated branch of my fork (up to date with ITHACA-FV/master)
